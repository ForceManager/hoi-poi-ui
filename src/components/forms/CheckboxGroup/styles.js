export default (theme) => ({
    root: {},
    Label: {
        width: 150,
        marginRight: 17,
        flexShrink: 0,
    },
    error: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.red500,
    },
    errorPerCheck: {
        ...theme.typography.smallText,
        color: theme.colors.red500,
        margin: '-6px 0 14px',
    },
    formControl: {
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        width: 290,
    },
    isFullWidth: {
        width: '100%',
    },
    checkboxWrapper: {
        display: 'flex',
        alignItems: 'center',
        marginBottom: 14,
        '&:hover $info': {
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
        },
    },
    checkboxControl: {
        display: 'flex',
        alignItems: 'center',
        lineHeight: '20px',
        cursor: 'pointer',
        '&:last-child': {
            marginBottom: 0,
        },
    },
    checkboxLabel: {
        marginLeft: 10,
        ...theme.typography.defaultText,
        color: theme.colors.neutral700,
    },
    iconContainer: {
        height: 18,
        width: 18,
        marginLeft: 5,
    },
    info: {
        display: 'none',
        width: '100%',
        height: '100%',
        cursor: 'pointer',
    },
    popoverTitle: {
        marginBottom: 5,
    },
    isReadOnly: {
        '& $formControl': {
            cursor: 'not-allowed',
            '&::before': {
                borderBottom: `1px solid ${theme.colors.neutral400}`,
            },
            '&::after': {
                borderBottom: `1px solid ${theme.colors.neutral400}`,
            },
        },
    },
    vertical: {
        display: 'block',
        padding: '8px 0',
        marginLeft: '0 !important',
        '& $formControl': {
            marginTop: 10,
        },
    },
    horizontal: {
        display: 'flex',
        alignItems: 'flex-start',
        padding: '8px 0',
    },
});
