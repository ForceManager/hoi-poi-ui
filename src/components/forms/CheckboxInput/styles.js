export default (theme) => ({
    root: {
        outline: 'none',
    },
    Label: {
        width: 153,
        marginRight: 17,
        padding: '0 0 5px 0',
    },
    error: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.red,
    },
    formControl: {
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
        width: 290,
        lineHeight: 0,
    },
    checkbox: {
        width: '100%',
        lineHeight: '20px',
        flexGrow: 1,
        outline: 'none',
        border: 'none',
    },
    errored: {
        paddingBottom: '30px !important',
        '& $formControl': {
            '&::before': {
                borderBottom: `1px solid ${theme.colors.red} !important`,
            },
            '&::after': {
                borderBottom: `1px solid ${theme.colors.red} !important`,
            },
        },
    },
    vertical: {
        display: 'block',
        padding: '15px 0 10px 0',
        marginLeft: '0 !important',
        '& $formControl': {
            marginTop: 10,
        },
        '& $Label': {
            width: '100%',
        },
    },
    horizontal: {
        display: 'flex',
        alignItems: 'center',
        padding: '15px 0 10px 0',
        '& $error': {},
    },
    isFullWidth: {
        '& $formControl': {
            width: '100%',
            flex: 1,
        },
        '&$vertical $Label': {
            width: '100%',
        },
    },
});
