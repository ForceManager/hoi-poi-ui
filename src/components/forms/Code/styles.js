export default (theme) => ({
    root: {
        outline: 'none',
        '& .ace_editor': {
            width: '100%',
        },
    },
    Label: {
        width: 150,
        marginRight: 17,
        padding: '0 0 8px 0',
        flexShrink: 0,
    },
    info: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.textLight.secondary,
    },
    error: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.semantic.negative500,
    },
    formControl: {
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: 320,
        lineHeight: 0,
    },
    isReadOnly: {},
    vertical: {
        display: 'block',
        padding: '15px 0 10px 0',
        marginLeft: '0 !important',
        '& $formControl': {
            marginTop: 10,
        },
        '& $Label': {
            width: '100%',
        },
    },
    horizontal: {
        display: 'flex',
        alignItems: 'flex-start',
        padding: '15px 0 10px 0',
    },
    isFullWidth: {
        '& $formControl': {
            width: '100%',
            flex: 1,
        },
        '&$vertical $Label': {
            width: '100%',
        },
    },
    focused: {},
});
