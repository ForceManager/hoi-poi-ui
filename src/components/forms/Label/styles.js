export default (theme) => ({
    root: {
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'center',
    },
    Text: {
        paddingRight: 5,
        ...theme.typography.body,
        color: theme.colors.neutral900,
    },
    info: {
        display: 'block',
        width: 10,
    },
    icon: {
        display: 'block',
        width: 13,
        height: 13,
        cursor: 'pointer',
        '& path': {
            fill: theme.colors.neutral600,
        },
    },
});
