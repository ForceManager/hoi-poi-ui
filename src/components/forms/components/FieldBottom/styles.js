export default (theme) => {
    return {
        root: {
            width: 320,
            marginTop: -3,
        },
        isFullWidth: {
            width: '100%',
        },
        info: {
            ...theme.typography.caption,
            color: theme.colors.grey[500],
            paddingBottom: 5,
        },
        error: {
            ...theme.typography.caption,
            color: theme.colors.red500,
            paddingBottom: 5,
        },
    };
};
