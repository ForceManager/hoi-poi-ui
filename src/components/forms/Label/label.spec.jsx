import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from 'utils/styles';
import Tooltip from 'components/forms/Section';
import Button from 'components/general/Button';

describe('Tooltip', () => {
    test('match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Tooltip placement="top" content={<span>Hello!</span>}>
                    <Button>top</Button>
                </Tooltip>
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('visible match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Tooltip placement="top" content={<span>Hello!</span>}>
                    <Button>top</Button>
                </Tooltip>
            </HoiPoiProvider>,
        );
        wrapper.simulate('mouseover');
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
