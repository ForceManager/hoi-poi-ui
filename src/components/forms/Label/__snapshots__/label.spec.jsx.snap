// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label default match 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.3",
        }
      }
    >
      <Label>
        <span
          className="HoiPoi__Label__root-1-0"
        >
          <Text
            className="HoiPoi__Label__Text-1-1"
            overrides={Object {}}
          >
            <span
              className="HoiPoi__Text__root-1-4 HoiPoi__Label__Text-1-1"
            >
              Lorem ipsum
            </span>
          </Text>
        </span>
      </Label>
    </JssProvider>
  </ThemeProvider>
</Component>
`;

exports[`Label is required match 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.3",
        }
      }
    >
      <Label
        isRequired={true}
      >
        <span
          className="HoiPoi__Label__root-2-0"
        >
          <Text
            className="HoiPoi__Label__Text-2-1"
            overrides={Object {}}
          >
            <span
              className="HoiPoi__Text__root-2-4 HoiPoi__Label__Text-2-1"
            >
              Lorem ipsum
              *
            </span>
          </Text>
        </span>
      </Label>
    </JssProvider>
  </ThemeProvider>
</Component>
`;

exports[`Label with hint match 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.3",
        }
      }
    >
      <Label
        hint="I am a hint"
      >
        <span
          className="HoiPoi__Label__root-3-0"
        >
          <Text
            className="HoiPoi__Label__Text-3-1"
            overrides={Object {}}
          >
            <span
              className="HoiPoi__Text__root-3-4 HoiPoi__Label__Text-3-1"
            >
              Lorem ipsum
            </span>
          </Text>
          <Popover
            content="I am a hint"
            overrides={Object {}}
            placement="top"
            trigger={
              Array [
                "hover",
              ]
            }
          >
            <Tooltip
              align={Object {}}
              arrowContent={null}
              destroyTooltipOnHide={false}
              mouseEnterDelay={0}
              mouseLeaveDelay={0.1}
              overlay="I am a hint"
              overlayClassName="HoiPoi__Popover__root-3-15"
              placement="top"
              prefixCls="hoi-poi-popover"
              transitionName="hoi-poi-popover--fade"
              trigger={
                Array [
                  "hover",
                ]
              }
            >
              <Trigger
                action={
                  Array [
                    "hover",
                  ]
                }
                afterPopupVisibleChange={[Function]}
                arrowContent={null}
                blurDelay={0.15}
                builtinPlacements={
                  Object {
                    "bottom": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tc",
                        "bc",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "bottomLeft": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "bottomRight": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tr",
                        "br",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "left": Object {
                      "offset": Array [
                        -4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "cr",
                        "cl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "leftBottom": Object {
                      "offset": Array [
                        -4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "br",
                        "bl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "leftTop": Object {
                      "offset": Array [
                        -4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tr",
                        "tl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "right": Object {
                      "offset": Array [
                        4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "cl",
                        "cr",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "rightBottom": Object {
                      "offset": Array [
                        4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bl",
                        "br",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "rightTop": Object {
                      "offset": Array [
                        4,
                        0,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "tr",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "top": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bc",
                        "tc",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "topLeft": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bl",
                        "tl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "topRight": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "br",
                        "tr",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                  }
                }
                defaultPopupVisible={false}
                destroyPopupOnHide={false}
                focusDelay={0}
                getDocument={[Function]}
                getPopupClassNameFromAlign={[Function]}
                hideAction={Array []}
                mask={false}
                maskClosable={true}
                mouseEnterDelay={0}
                mouseLeaveDelay={0.1}
                onPopupAlign={[Function]}
                onPopupVisibleChange={[Function]}
                overlay="I am a hint"
                popup={[Function]}
                popupAlign={Object {}}
                popupClassName="HoiPoi__Popover__root-3-15"
                popupPlacement="top"
                popupStyle={Object {}}
                popupTransitionName="hoi-poi-popover--fade"
                prefixCls="hoi-poi-popover"
                showAction={Array []}
              >
                <span
                  className="HoiPoi__Label__info-3-2"
                  key="trigger"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <Icon
                    className="HoiPoi__Label__icon-3-3"
                    name="info"
                    overrides={Object {}}
                    size="medium"
                  >
                    <span
                      className="HoiPoi__Icon__root-3-16 HoiPoi__Icon__medium-3-18 HoiPoi__Label__icon-3-3"
                    >
                      <Info
                        color="#788590"
                      >
                        <svg
                          viewBox="0 0 18 18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8.998.001C3.913.08-.077 4.227.001 9.157.08 14.165 3.991 18 8.841 18h.157c2.426-.078 4.694-1.017 6.416-2.739 1.72-1.722 2.66-3.991 2.581-6.417-.078-5.008-3.99-8.921-8.997-8.843zm1.096 13.93c0 .547-.47 1.095-1.096 1.095-.547 0-1.095-.47-1.095-1.095v-5.4c0-.548.47-1.096 1.095-1.096.548 0 1.096.47 1.096 1.096v5.4zm-1.096-7.2c-.704 0-1.33-.626-1.33-1.33 0-.704.626-1.252 1.33-1.252.704 0 1.33.626 1.33 1.33-.078.704-.626 1.252-1.33 1.252z"
                            fill="#788590"
                            fillRule="nonzero"
                          />
                        </svg>
                      </Info>
                    </span>
                  </Icon>
                </span>
              </Trigger>
            </Tooltip>
          </Popover>
        </span>
      </Label>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
